mitodl/ol-django
# Adds IP-based geolocation app

# What are the relevant tickets?

Part of https://github.com/mitodl/hq/issues/2443

# Description (What does it do?)

For https://github.com/mitodl/hq/issues/2443, we developed a Django app that provides geolocation via IP address for clients that does not rely on a third-party API service and instead uses a local copy of the MaxMind GeoIP database. (There are a bunch of these that exist now but they all tend to require that you hit an external API, and I wanted to avoid that.) While this was developed within xPRO, it was always intended to be pulled out into a library we could use elsewhere - this is the library version of the same code. 

# How can this be tested?

Automated tests should pass. 

For local testing, you will need to import the MaxMind GeoLite2 databases as described in the README. Then, you can import the `ip_to_country_code` API and run it against most any IP address. You should get an ISO country code back. 

Note that MaxMind does not store data for the three private IPv4 network ranges - there's a management command to create blocks for this for local development use.

# Additional Context

The GeoIP2 and GeoLite2 datasets are interchangeable but I recommend using the GeoLite2 database as you can get it for free. The code here only resolves down to the country so, while there are more complete datasets available from MaxMind, using them won't help too much unless you want to use the database for other things. 

We'll need to use this when/if we have to add tax collection to MITx Online. The API is the same as it is in xPRO so once this is released a PR should go up to have xPRO use this instead.
